common/app/routes/challenges/components/Classic-Modal.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { PropTypes } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import PureComponent from 'react-pure-render/component';
+import FontAwesome from 'react-fontawesome';
+
+export default class ClassicModal extends PureComponent {
+
+  onClickHandler() {
+    this.props.onModalClick();
+  }
+
+  renderIcon() {
+    return (
+      <FontAwesome
+        name='check-circle'
+      />
+    );
+  }
+
+  render() {
+    let {
+      keyBindings,
+      modalShow,
+      modalTitle,
+      onModalExit
+    } = this.props;
+
+    return (
+      <Modal
+      animation={ false }
+      dialogClassName='success'
+      keyboard={ true }
+      onHide={ onModalExit }
+      onKeyDown={ keyBindings }
+      onKeyUp={ keyBindings }
+      show={ modalShow }
+      >
+        <Modal.Header className='challenge-list-header' closeButton={ true }>
+          <Modal.Title>{ modalTitle }</Modal.Title>
+        </Modal.Header>
+        <Modal.Body className='completion-icon text-primary'>
+          { this.renderIcon() }
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+          block={ true }
+          bsSize='large'
+          bsStyle='primary'
+          onClick={this.onClickHandler.bind(this)}
```

Comment:
Pass the function prop directly. Using the bind method is the same as creating a new function per render.